src/fake-timers-src.js

```diff
@@ -1903,21 +1913,30 @@ function withGlobal(_global) {
                             };
 
                             const handle = clock.setTimeout(() => {
-                                options.signal?.removeEventListener(
-                                    "abort",
-                                    abort,
-                                );
+                                if (options.signal) {
```

Comment:
Opted to move back to regular if checks as opposed to optional chaining for consistency and readability, as you can't optional-chain adding / removing signals to `abortListenerMap`